indico/modules/events/editing/service.py

```diff
@@ -148,3 +137,65 @@ def service_handle_new_editable(editable):
     except requests.RequestException as exc:
         logger.exception('Failed calling listener for editable')
         raise ServiceRequestFailed(exc)
+
+
+def service_handle_review_editable(editable, action, parent_revision, revision=None):
+    new_revision = revision or parent_revision
+    data = {
+        'action': action.name,
+        'revision': EditingRevisionUnclaimedSchema().dump(new_revision),
+        'endpoints': _get_revision_endpoints(new_revision)
+    }
+    try:
+        path = '/event/{}/editable/{}/{}/{}'.format(
+            _get_event_identifier(editable.event),
+            editable.type.name,
+            editable.contribution_id,
+            new_revision.id
+        )
+        resp = requests.post(_build_url(editable.event, path), headers=_get_headers(editable.event),
+                             json=data)
+        resp.raise_for_status()
+        resp = ServiceReviewEditableSchema().load(resp.json())
+
+        if 'comment' in resp:
+            parent_revision.comment = resp.get('comment')
+        if 'tags' in resp:
+            parent_revision.tags = {tag for tag in editable.event.editing_tags
+                                    if tag.id in map(int, resp.get('tags'))}
+        if 'comments' in resp:
+            for comment in resp.get('comments'):
+                create_revision_comment(new_revision, User.get_system_user(), comment.get('text'),
+                                        internal=comment.get('internal'))
```

Comment:
only use `.get()` if you expect something to not exist. but if fields in the schema are required or have a default (`missing=...`) value those keys always exist

```suggestion
        for comment in resp.get('comments', []):
            create_revision_comment(new_revision, User.get_system_user(), comment['text'],
                                    internal=comment['internal'])
```

Reply:
For some reason, with dictionaries, I find it more idiomatic than pure parentheses, must be from Java.
But good catch, didn't expect comments to be undefined at this point.